app/components/sidebars/settings/settings_sidebar_base.js

```diff
@@ -19,33 +19,111 @@ import {t} from '@utils/i18n';
+            unsetCustomStatus: PropTypes.func.isRequired,
         }).isRequired,
         currentUser: PropTypes.object.isRequired,
         status: PropTypes.string,
         theme: PropTypes.object.isRequired,
+        isCustomStatusEnabled: PropTypes.bool.isRequired,
+        customStatus: PropTypes.object,
+        setStatusRequestStatus: PropTypes.string,
+        clearStatusRequestStatus: PropTypes.string,
     };
 
     static defaultProps = {
         currentUser: {},
         status: 'offline',
+        customStatus: {},
     };
 
+    constructor(props) {
+        super(props);
+        this.state = {
+            showStatus: true,
+            showRetryMessage: false,
+        };
+    }
+
     componentDidMount() {
         this.mounted = true;
         EventEmitter.on(NavigationTypes.CLOSE_SETTINGS_SIDEBAR, this.closeSettingsSidebar);
     }
 
+    componentDidUpdate(prevProps) {
+        const {setStatusRequestStatus, clearStatusRequestStatus, customStatus} = this.props;
+
+        if (prevProps.clearStatusRequestStatus !== clearStatusRequestStatus) {
+            this.handleClearStatusRequestStatusChange();
+        }
+
+        if (prevProps.setStatusRequestStatus !== setStatusRequestStatus) {
+            this.handleSetStatusRequestStatusChange();
+        }
+
+        if (prevProps.clearStatusRequestStatus === clearStatusRequestStatus && prevProps.setStatusRequestStatus === setStatusRequestStatus) {
+            this.handleCustomStatusChange(prevProps.customStatus, customStatus);
+        }
+    }
+
     componentWillUnmount() {
         this.mounted = false;
         EventEmitter.off(NavigationTypes.CLOSE_SETTINGS_SIDEBAR, this.closeSettingsSidebar);
     }
 
+    handleClearStatusRequestStatusChange = () => {
+        const {clearStatusRequestStatus} = this.props;
+        if (clearStatusRequestStatus === RequestStatus.STARTED || clearStatusRequestStatus === RequestStatus.SUCCESS) {
+            this.setState({
+                showStatus: false,
+                showRetryMessage: false,
+            });
+        } else if (clearStatusRequestStatus === RequestStatus.FAILURE) {
+            this.setState({
+                showStatus: true,
+                showRetryMessage: true,
+            });
+        }
+    }
+
+    handleSetStatusRequestStatusChange = () => {
+        const {setStatusRequestStatus} = this.props;
+        if (setStatusRequestStatus === RequestStatus.STARTED || setStatusRequestStatus === RequestStatus.SUCCESS) {
+            this.setState({
+                showStatus: true,
+                showRetryMessage: false,
+            });
+        } else if (setStatusRequestStatus === RequestStatus.FAILURE) {
+            this.setState({
+                showStatus: true,
+                showRetryMessage: true,
+            });
+        }
+    }
+
+    handleCustomStatusChange = (prevCustomStatus, customStatus) => {
+        const isStatusSet = customStatus.emoji;
```

Comment:
variables that will be binary (`is*`) should be booleans.

```suggestion
        const isStatusSet = Boolean(customStatus.emoji);
```